Apply Debian-stable patch: Fix memory corruption

h3xx opened the pull request:
Source: https://sources.debian.org/patches/xautolock/1:2.2-5.1/10-fix-memory-corruption.patch/

l0b0 commented:
Thank you very much for all these patches, @h3xx! This is quite a surprise – I can't even recall why I made this repo over four years ago, other than perhaps wanting to build it myself. My "contribution" is simply making a copy of the original sources, possibly with a vague hope that this great little piece of software might end up with a version history. Since then I've given up on maintaining my own desktop from scratch, so I'm no longer using `xautolock`. Also, I'm in no way competent to review these patches. I'm familiar with several languages, but I've only touched C for a few days many years back.

In summary, I'm not a contributor, distributor or user. I can't even ask you to take over this project, as I don't even know whether anyone is still maintaining it. I'm going to archive it soon, but if you have any suggestions feel free to let me know here.

stigge committed:
Fix memory corruption (Debian patch)

Description: Fix memory corruption
 xautolock: Uses freed memory for starting the locker
Original Author: Roland Stigge <stigge@antcom.de>
Bug-Debian: http://bugs.debian.org/cgi-bin/bugreport.cgi?bug=514406
